modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1759,75 +1761,91 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
         }
 
         int partId = extractPartitionNumber(pendingAssignmentsEntry.key());
-        int tblId = extractTableId(pendingAssignmentsEntry.key(), PENDING_ASSIGNMENTS_PREFIX);
+        int zoneId = extractZoneId(pendingAssignmentsEntry.key(), PENDING_ASSIGNMENTS_PREFIX);
 
-        var replicaGrpId = new TablePartitionId(tblId, partId);
+        var zonePartitionId = new ZonePartitionId(zoneId, partId);
 
         // Stable assignments from the meta store, which revision is bounded by the current pending event.
-        Entry stableAssignmentsEntry = metaStorageMgr.getLocally(stablePartAssignmentsKey(replicaGrpId), revision);
+        Entry stableAssignmentsEntry = metaStorageMgr.getLocally(stablePartAssignmentsKey(zonePartitionId), revision);
```

Comment:
Offtopic, but isn't a good idea to move all `get*Assignments*` methods to idk AssignmentsManager with metastore inside?